src/sdk/cohort.ts

```diff
@@ -0,0 +1,73 @@
+import { Porter } from '../characters/porter';
+import { ChecksumAddress } from '../types';
+
+export type CohortConfiguration = {
+  readonly threshold: number;
+  readonly shares: number;
+  readonly porterUri: string;
+};
+
+export type CohortJSON = {
+  ursulaAddresses: ChecksumAddress[];
+  threshold: number;
+  shares: number;
+  porterUri: string;
+};
+
+export class Cohort {
+  private constructor(
+    public ursulaAddresses: ChecksumAddress[],
+    public readonly configuration: CohortConfiguration
+  ) {}
+
+  public static async create(
+    configuration: CohortConfiguration,
+    include = [],
+    exclude = []
+  ) {
+    const porter = new Porter(configuration.porterUri);
+    const ursulas = await porter.getUrsulas(
+      configuration.shares,
+      exclude,
+      include
+    );
+    const ursulaAddresses = ursulas.map((ursula) => ursula.checksumAddress);
+    return new Cohort(ursulaAddresses, configuration);
+  }
+
+  public get shares(): number {
+    return this.ursulaAddresses.length;
+  }
+
+  public toJSON() {
+    return JSON.stringify(this.toObj());
+  }
+
+  public static fromJSON(json: string) {
+    const config = JSON.parse(json);
+    return Cohort.fromObj(config);
+  }
+
+  public static fromObj({
+    ursulaAddresses,
+    threshold,
+    shares,
+    porterUri,
+  }: CohortJSON) {
+    const config = {
+      threshold: threshold,
+      shares: shares,
+      porterUri: porterUri,
+    };
+    return new Cohort(ursulaAddresses, config);
+  }
+
+  public toObj(): CohortJSON {
```

Comment:
Yes, we don't have to specify returns objects; this type is used primarily internally. It's useful for type-checking when using common TypeScript idioms, such as deconstructing, etc.

I think we should expose typed objects instead of using opaque types (dicts, objects, JSON) when we can because it significantly improves the experience for TS developers.